File: User-Service-RH-/src/main/java/com/example/demo/models/Formation.java
package com.example.demo.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import java.sql.Timestamp;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Formation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String nom;
    private Timestamp dateDebut;
    private Timestamp dateFin;
    private String domaine;
    private int nbPlaceDisponible;
    private String nomSalle;
    private Boolean etatFormation;


}

File: src/app/shared/Formation.ts
export class Formation {
  id :any  ;
  nom :string="" ;
  dateDebut:any;
  dateFin:any ;
  domaine:string ="" ;
  nbPlaceDisponible:any   ;
  nomSalle:string  ="" ;
  etatFormation:any ;


  Formation(nom :string  ,formerName :string  ,  dateDebut:any, dateFin:any, domaine:string ,  nbPlaceDisponible: number,nomSalle:string, etatFormation:boolean  , id: number )
  {
    this.id=id;
    this.nom=nom ;
    this.dateDebut=dateDebut ;
    this.dateFin=dateFin ;
    this.domaine=domaine ;
    this.nbPlaceDisponible=nbPlaceDisponible ;
    this.nomSalle=nomSalle ;
    this.etatFormation=etatFormation ;

  }

}
